// @coinbase/onchainkit/esm/identity/utils/getName.js
import { base, mainnet } from 'viem/chains';
import { getChainPublicClient } from '../../core/network/getChainPublicClient.js';
import { isBase } from '../../core/utils/isBase.js';
import { isEthereum } from '../../core/utils/isEthereum.js';
import L2ResolverAbi from '../abis/L2ResolverAbi.js';
import { RESOLVER_ADDRESSES_BY_CHAIN_ID } from '../constants.js';
import { convertReverseNodeToBytes } from './convertReverseNodeToBytes.js';

/**
 * An asynchronous function to fetch the Ethereum Name Service (ENS)
 * name for a given Ethereum address. It returns the ENS name if it exists,
 * or null if it doesn't or in case of an error.
 */
const getName = async ({
  address,
  chain = mainnet
}) => {
  const chainIsBase = isBase({
    chainId: chain.id
  });
  const chainIsEthereum = isEthereum({
    chainId: chain.id
  });
  const chainSupportsUniversalResolver = chainIsEthereum || chainIsBase;
  if (!chainSupportsUniversalResolver) {
    return Promise.reject('ChainId not supported, name resolution is only supported on Ethereum and Base.');
  }
  const client = getChainPublicClient(chain);
  if (chainIsBase) {
    const addressReverseNode = convertReverseNodeToBytes(address, base.id);
    try {
      const basename = await client.readContract({
        abi: L2ResolverAbi,
        address: RESOLVER_ADDRESSES_BY_CHAIN_ID[chain.id],
        functionName: 'name',
        args: [addressReverseNode]
      });
      if (basename) {
        return basename;
      }
    } catch (_error) {
      // This is a best effort attempt, so we don't need to do anything here.
    }
  }

  // ENS username
  const ensName = await client.getEnsName({
    address
  });
  return ensName ?? null;
};
export { getName };
