// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedContent.js
import { cn, background, border, text } from '../../styles/theme.js';
import { WALLET_ADVANCED_DEFAULT_SWAPPABLE_TOKENS } from '../constants.js';
import { useWalletAdvancedContext } from './WalletAdvancedProvider.js';
import { WalletAdvancedQrReceive } from './WalletAdvancedQrReceive.js';
import { WalletAdvancedSwap } from './WalletAdvancedSwap.js';
import { useWalletContext } from './WalletProvider.js';
import { jsxs, jsx } from 'react/jsx-runtime';
function WalletAdvancedContent({
  children,
  swappableTokens
}) {
  const _useWalletContext = useWalletContext(),
    isSubComponentClosing = _useWalletContext.isSubComponentClosing,
    setIsSubComponentOpen = _useWalletContext.setIsSubComponentOpen,
    setIsSubComponentClosing = _useWalletContext.setIsSubComponentClosing;
  const _useWalletAdvancedCon = useWalletAdvancedContext(),
    showQr = _useWalletAdvancedCon.showQr,
    showSwap = _useWalletAdvancedCon.showSwap,
    tokenBalances = _useWalletAdvancedCon.tokenBalances,
    animations = _useWalletAdvancedCon.animations;
  return /*#__PURE__*/jsxs("div", {
    "data-testid": "ockWalletAdvancedContent",
    className: cn(background.default, border.radius, border.lineDefault, 'my-1.5 h-auto w-88', 'flex items-center justify-center', animations.container),
    onAnimationEnd: () => {
      if (isSubComponentClosing) {
        setIsSubComponentOpen(false);
        setIsSubComponentClosing(false);
      }
    },
    children: [/*#__PURE__*/jsx("div", {
      className: cn('flex flex-col items-center justify-center', 'h-120 w-88', showQr ? '' : 'hidden'),
      children: /*#__PURE__*/jsx(WalletAdvancedQrReceive, {})
    }), /*#__PURE__*/jsx("div", {
      className: cn('flex flex-col items-center justify-center', 'h-120 w-88', showSwap ? '' : 'hidden'),
      children: /*#__PURE__*/jsx(WalletAdvancedSwap, {
        title: /*#__PURE__*/jsx("div", {
          className: cn(text.headline, 'w-full text-center text-base'),
          children: "Swap"
        }),
        to: swappableTokens ?? WALLET_ADVANCED_DEFAULT_SWAPPABLE_TOKENS,
        from: tokenBalances?.map(token => ({
          address: token.address,
          chainId: token.chainId,
          symbol: token.symbol,
          decimals: token.decimals,
          image: token.image,
          name: token.name
        })) ?? [],
        className: "w-full px-4 pt-3 pb-4"
      })
    }), /*#__PURE__*/jsx("div", {
      className: cn('flex flex-col items-center justify-between', 'h-120 w-88', 'px-4 py-3', showQr || showSwap ? 'hidden' : ''),
      children: children
    })]
  });
}
export { WalletAdvancedContent };
