// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedProvider.js
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
import { useValue } from '../../internal/hooks/useValue.js';
import { usePortfolio } from '../hooks/usePortfolio.js';
import { createContext, useContext, useState } from 'react';
import { useWalletContext } from './WalletProvider.js';
import { jsx } from 'react/jsx-runtime';
const emptyContext = {};
const WalletAdvancedContext = /*#__PURE__*/createContext(emptyContext);
function useWalletAdvancedContext() {
  const walletAdvancedContext = useContext(WalletAdvancedContext);
  if (walletAdvancedContext === emptyContext) {
    throw new Error('useWalletAdvancedContext must be used within a WalletAdvancedProvider');
  }
  return walletAdvancedContext;
}
function WalletAdvancedProvider({
  children
}) {
  const _useWalletContext = useWalletContext(),
    address = _useWalletContext.address,
    isSubComponentClosing = _useWalletContext.isSubComponentClosing,
    showSubComponentAbove = _useWalletContext.showSubComponentAbove;
  const _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    showSwap = _useState2[0],
    setShowSwap = _useState2[1];
  const _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isSwapClosing = _useState4[0],
    setIsSwapClosing = _useState4[1];
  const _useState5 = useState(false),
    _useState6 = _slicedToArray(_useState5, 2),
    showQr = _useState6[0],
    setShowQr = _useState6[1];
  const _useState7 = useState(false),
    _useState8 = _slicedToArray(_useState7, 2),
    isQrClosing = _useState8[0],
    setIsQrClosing = _useState8[1];
  const _usePortfolio = usePortfolio({
      address
    }),
    portfolioData = _usePortfolio.data,
    refetchPortfolioData = _usePortfolio.refetch,
    isFetchingPortfolioData = _usePortfolio.isFetching,
    portfolioDataUpdatedAt = _usePortfolio.dataUpdatedAt;
  const portfolioFiatValue = portfolioData?.portfolioBalanceInUsd;
  const tokenBalances = portfolioData?.tokenBalances;
  const animations = getAnimations(isSubComponentClosing, showSubComponentAbove);
  const value = useValue({
    showSwap,
    setShowSwap,
    isSwapClosing,
    setIsSwapClosing,
    showQr,
    setShowQr,
    isQrClosing,
    setIsQrClosing,
    tokenBalances,
    portfolioFiatValue,
    isFetchingPortfolioData,
    portfolioDataUpdatedAt,
    refetchPortfolioData,
    animations
  });
  return /*#__PURE__*/jsx(WalletAdvancedContext.Provider, {
    value: value,
    children: children
  });
}
function getAnimations(isSubComponentClosing, showSubComponentAbove) {
  if (isSubComponentClosing) {
    return {
      container: showSubComponentAbove ? 'fade-out slide-out-to-bottom-1.5 animate-out fill-mode-forwards ease-in-out' : 'fade-out slide-out-to-top-1.5 animate-out fill-mode-forwards ease-in-out',
      content: ''
    };
  }
  return {
    container: showSubComponentAbove ? 'fade-in slide-in-from-bottom-1.5 animate-in duration-300 ease-out' : 'fade-in slide-in-from-top-1.5 animate-in duration-300 ease-out',
    content: showSubComponentAbove ? 'fade-in slide-in-from-bottom-2.5 animate-in fill-mode-forwards duration-300 ease-out' : 'fade-in slide-in-from-top-2.5 animate-in fill-mode-forwards duration-300 ease-out'
  };
}
export { WalletAdvancedProvider, useWalletAdvancedContext };
