// @coinbase/onchainkit/esm/wallet/components/Wallet.js
'use client';
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
import { Draggable } from '../../internal/components/Draggable/Draggable.js';
import { useIsMounted } from '../../internal/hooks/useIsMounted.js';
import { useOutsideClick } from '../../internal/hooks/useOutsideClick.js';
import { useTheme } from '../../internal/hooks/useTheme.js';
import { findComponent } from '../../internal/utils/findComponent.js';
import { cn } from '../../styles/theme.js';
import { useRef, useMemo, Children } from 'react';
import { getWalletDraggableProps } from '../utils/getWalletDraggableProps.js';
import { ConnectWallet } from './ConnectWallet.js';
import { WalletAdvanced } from './WalletAdvanced.js';
import { WalletDropdown } from './WalletDropdown.js';
import { WalletProvider, useWalletContext } from './WalletProvider.js';
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
const Wallet = ({
  children,
  className,
  draggable,
  draggableStartingPosition
}) => {
  const componentTheme = useTheme();
  const isMounted = useIsMounted();
  // prevents SSR hydration issue
  if (!isMounted) {
    return null;
  }
  return /*#__PURE__*/jsx(WalletProvider, {
    children: /*#__PURE__*/jsx(WalletContent, _objectSpread(_objectSpread({
      className: cn(componentTheme, className)
    }, getWalletDraggableProps({
      draggable,
      draggableStartingPosition
    })), {}, {
      children: children
    }))
  });
};
function WalletContent({
  children,
  className,
  draggable,
  draggableStartingPosition
}) {
  const _useWalletContext = useWalletContext(),
    isSubComponentOpen = _useWalletContext.isSubComponentOpen,
    isConnectModalOpen = _useWalletContext.isConnectModalOpen,
    handleClose = _useWalletContext.handleClose,
    connectRef = _useWalletContext.connectRef,
    showSubComponentAbove = _useWalletContext.showSubComponentAbove,
    alignSubComponentRight = _useWalletContext.alignSubComponentRight;
  const walletContainerRef = useRef(null);
  useOutsideClick(walletContainerRef, handleClose);
  const _useMemo = useMemo(() => {
      const childrenArray = Children.toArray(children);
      return {
        connect: childrenArray.find(findComponent(ConnectWallet)),
        dropdown: childrenArray.find(findComponent(WalletDropdown)),
        advanced: childrenArray.find(findComponent(WalletAdvanced))
      };
    }, [children]),
    connect = _useMemo.connect,
    dropdown = _useMemo.dropdown,
    advanced = _useMemo.advanced;
  if (dropdown && advanced) {
    console.error('Defaulted to WalletDropdown. Wallet cannot have both WalletDropdown and WalletAdvanced as children.');
  }
  if (draggable) {
    return /*#__PURE__*/jsx("div", {
      ref: walletContainerRef,
      className: cn('relative w-fit shrink-0', className),
      children: /*#__PURE__*/jsx(Draggable, {
        startingPosition: draggableStartingPosition,
        disabled: isConnectModalOpen,
        children: /*#__PURE__*/jsx(WalletSubComponent, {
          connect: connect,
          connectRef: connectRef,
          dropdown: dropdown,
          advanced: advanced,
          isSubComponentOpen: isSubComponentOpen,
          alignSubComponentRight: alignSubComponentRight,
          showSubComponentAbove: showSubComponentAbove
        })
      })
    });
  }
  return /*#__PURE__*/jsx("div", {
    ref: walletContainerRef,
    className: cn('relative w-fit shrink-0', className),
    children: /*#__PURE__*/jsx(WalletSubComponent, {
      connect: connect,
      connectRef: connectRef,
      dropdown: dropdown,
      advanced: advanced,
      isSubComponentOpen: isSubComponentOpen,
      alignSubComponentRight: alignSubComponentRight,
      showSubComponentAbove: showSubComponentAbove
    })
  });
}
function WalletSubComponent({
  connect,
  connectRef,
  dropdown,
  advanced,
  isSubComponentOpen,
  alignSubComponentRight,
  showSubComponentAbove
}) {
  if (dropdown) {
    return /*#__PURE__*/jsxs(Fragment, {
      children: [connect, isSubComponentOpen && dropdown]
    });
  }
  return /*#__PURE__*/jsxs(Fragment, {
    children: [/*#__PURE__*/jsx("div", {
      ref: connectRef,
      children: connect
    }), isSubComponentOpen && /*#__PURE__*/jsx("div", {
      "data-testid": "ockWalletAdvancedContainer",
      className: cn('absolute', showSubComponentAbove ? 'bottom-full' : 'top-full', alignSubComponentRight ? 'right-0' : 'left-0'),
      children: advanced
    })]
  });
}
export { Wallet };
