// @coinbase/onchainkit/esm/wallet/components/WalletAdvancedSwap.js
'use client';
import { PressableIcon } from '../../internal/components/PressableIcon.js';
import { backArrowSvg } from '../../internal/svg/backArrowSvg.js';
import { cn } from '../../styles/theme.js';
import { Swap } from '../../swap/components/Swap.js';
import { SwapAmountInput } from '../../swap/components/SwapAmountInput.js';
import { SwapButton } from '../../swap/components/SwapButton.js';
import { SwapMessage } from '../../swap/components/SwapMessage.js';
import { SwapSettings } from '../../swap/components/SwapSettings.js';
import { SwapSettingsSlippageDescription } from '../../swap/components/SwapSettingsSlippageDescription.js';
import { SwapSettingsSlippageInput } from '../../swap/components/SwapSettingsSlippageInput.js';
import { SwapSettingsSlippageTitle } from '../../swap/components/SwapSettingsSlippageTitle.js';
import { SwapToast } from '../../swap/components/SwapToast.js';
import { SwapToggleButton } from '../../swap/components/SwapToggleButton.js';
import { useCallback } from 'react';
import { useWalletAdvancedContext } from './WalletAdvancedProvider.js';
import { jsx, jsxs } from 'react/jsx-runtime';
function WalletAdvancedSwap({
  config,
  className,
  disabled,
  experimental,
  from,
  isSponsored = false,
  onError,
  onStatus,
  onSuccess,
  title,
  to
}) {
  const _useWalletAdvancedCon = useWalletAdvancedContext(),
    setShowSwap = _useWalletAdvancedCon.setShowSwap,
    isSwapClosing = _useWalletAdvancedCon.isSwapClosing,
    setIsSwapClosing = _useWalletAdvancedCon.setIsSwapClosing;
  const handleCloseSwap = useCallback(() => {
    setIsSwapClosing(true);
  }, [setIsSwapClosing]);
  const handleAnimationEnd = useCallback(() => {
    if (isSwapClosing) {
      setShowSwap(false);
      setIsSwapClosing(false);
    }
  }, [isSwapClosing, setShowSwap, setIsSwapClosing]);
  const backButton = /*#__PURE__*/jsx(PressableIcon, {
    ariaLabel: "Back button",
    onClick: handleCloseSwap,
    children: /*#__PURE__*/jsx("div", {
      className: "p-2",
      children: backArrowSvg
    })
  });
  return /*#__PURE__*/jsx("div", {
    className: cn('h-full w-full', isSwapClosing ? 'fade-out slide-out-to-right-5 animate-out fill-mode-forwards ease-in-out' : 'fade-in slide-in-from-right-5 linear animate-in duration-150', 'relative'),
    onAnimationEnd: handleAnimationEnd,
    "data-testid": "ockWalletAdvancedSwap",
    children: /*#__PURE__*/jsxs(Swap, {
      className: className,
      onStatus: onStatus,
      onSuccess: onSuccess,
      onError: onError,
      config: config,
      isSponsored: isSponsored,
      title: title,
      experimental: experimental,
      headerLeftContent: backButton,
      children: [/*#__PURE__*/jsxs(SwapSettings, {
        className: "w-auto",
        children: [/*#__PURE__*/jsx(SwapSettingsSlippageTitle, {
          children: "Max. slippage"
        }), /*#__PURE__*/jsx(SwapSettingsSlippageDescription, {
          children: "Your swap will revert if the prices change by more than the selected percentage."
        }), /*#__PURE__*/jsx(SwapSettingsSlippageInput, {})]
      }), /*#__PURE__*/jsx(SwapAmountInput, {
        label: "Sell",
        swappableTokens: from,
        type: "from"
      }), /*#__PURE__*/jsx(SwapToggleButton, {}), /*#__PURE__*/jsx(SwapAmountInput, {
        label: "Buy",
        swappableTokens: to,
        type: "to"
      }), /*#__PURE__*/jsx(SwapButton, {
        disabled: disabled
      }), /*#__PURE__*/jsx(SwapMessage, {}), /*#__PURE__*/jsx(SwapToast, {})]
    })
  });
}
export { WalletAdvancedSwap };
