// @coinbase/onchainkit/esm/fund/components/FundCardPaymentMethodSelectRow.js
import { memo, useCallback } from 'react';
import { cn, pressable, border, background, text, color } from '../../styles/theme.js';
import { FundCardPaymentMethodImage } from './FundCardPaymentMethodImage.js';
import { jsx, jsxs } from 'react/jsx-runtime';
const FundCardPaymentMethodSelectRow = /*#__PURE__*/memo(({
  paymentMethod,
  onClick,
  hideImage,
  hideDescription,
  disabled,
  disabledReason,
  testId
}) => {
  const handleOnClick = useCallback(() => !disabled && onClick?.(paymentMethod), [disabled, onClick, paymentMethod]);
  return /*#__PURE__*/jsx("button", {
    "data-testid": testId,
    type: "button",
    className: cn(pressable.default, border.radius, background.default, 'flex w-full items-center justify-between px-4 py-2', {
      [pressable.disabled]: disabled
    }),
    onClick: handleOnClick,
    disabled: disabled,
    title: disabledReason,
    children: /*#__PURE__*/jsxs("span", {
      className: "flex items-center gap-3",
      children: [!hideImage && /*#__PURE__*/jsx(FundCardPaymentMethodImage, {
        paymentMethod: paymentMethod,
        className: cn('h-4 w-4', {
          [pressable.disabled]: disabled
        })
      }), /*#__PURE__*/jsxs("span", {
        className: "flex flex-col items-start",
        children: [/*#__PURE__*/jsx("span", {
          className: cn(text.headline),
          children: paymentMethod.name
        }), !hideDescription && /*#__PURE__*/jsx("span", {
          className: cn(text.label2, color.foregroundMuted, 'font-normal'),
          children: disabledReason || paymentMethod.description
        })]
      })]
    })
  });
});
export { FundCardPaymentMethodSelectRow };
