// @coinbase/onchainkit/esm/fund/components/FundCardAmountInputTypeSwitch.js
import { formatFiatAmount } from '../../internal/utils/formatFiatAmount.js';
import { truncateDecimalPlaces } from '../../internal/utils/truncateDecimalPlaces.js';
import { useCallback, useMemo } from 'react';
import { Skeleton } from '../../internal/components/Skeleton.js';
import { useIcon } from '../../internal/hooks/useIcon.js';
import { cn, text, pressable } from '../../styles/theme.js';
import { useFundContext } from './FundCardProvider.js';
import { jsx, jsxs } from 'react/jsx-runtime';
const FundCardAmountInputTypeSwitch = ({
  className
}) => {
  const _useFundContext = useFundContext(),
    selectedInputType = _useFundContext.selectedInputType,
    setSelectedInputType = _useFundContext.setSelectedInputType,
    asset = _useFundContext.asset,
    fundAmountFiat = _useFundContext.fundAmountFiat,
    fundAmountCrypto = _useFundContext.fundAmountCrypto,
    exchangeRate = _useFundContext.exchangeRate,
    exchangeRateLoading = _useFundContext.exchangeRateLoading,
    currency = _useFundContext.currency;
  const iconSvg = useIcon({
    icon: 'toggle'
  });
  const handleToggle = useCallback(() => {
    setSelectedInputType(selectedInputType === 'fiat' ? 'crypto' : 'fiat');
  }, [selectedInputType, setSelectedInputType]);
  const formatCrypto = useCallback(amount => {
    return `${truncateDecimalPlaces(amount || '0', 8)} ${asset}`;
  }, [asset]);
  const amountLine = useMemo(() => {
    return /*#__PURE__*/jsx("span", {
      "data-testid": "ockAmountLine",
      className: cn(text.label1),
      children: selectedInputType === 'fiat' ? formatCrypto(fundAmountCrypto) : formatFiatAmount({
        amount: fundAmountFiat,
        currency: currency,
        minimumFractionDigits: 0
      })
    });
  }, [fundAmountCrypto, fundAmountFiat, selectedInputType, formatCrypto, currency]);
  if (exchangeRateLoading || !exchangeRate) {
    return /*#__PURE__*/jsx(Skeleton, {
      className: "h-[1.625rem]"
    });
  }
  return /*#__PURE__*/jsxs("div", {
    className: cn('flex items-center', className),
    children: [/*#__PURE__*/jsx("button", {
      type: "button",
      "aria-label": "amount type switch",
      className: cn(pressable.default, 'mr-1 rounded-full p-1 opacity-50 transition-opacity hover:opacity-100'),
      "data-testid": "ockAmountTypeSwitch",
      onClick: handleToggle,
      children: /*#__PURE__*/jsx("div", {
        className: "h-[1.125rem] w-[1.125rem]",
        children: iconSvg
      })
    }), /*#__PURE__*/jsx("div", {
      className: "w-full truncate",
      children: amountLine
    })]
  });
};
export { FundCardAmountInputTypeSwitch, FundCardAmountInputTypeSwitch as default };
